graphql2/schema.graphql

```diff
@@ -245,19 +245,28 @@ type MessageLogConnection {
   stats: MessageLogConnectionStats!
 }
 
+enum MessageLogSegmentBy {
+  service
+  messageType
+  user
+  # Add other segments specific to message logs here
+}
+
 type MessageLogConnectionStats {
   timeSeries(input: TimeSeriesOptions!): [TimeSeriesBucket!]!
 }
 
 input TimeSeriesOptions {
   bucketDuration: ISODuration!
   bucketOrigin: ISOTimestamp
+  segmentBy: MessageLogSegmentBy
```

Comment:
Since `MessageLogSegmentBy` is message-log specific, let's add it as an optional parameter to `timeSeries` instead

```graphql
type MessageLogConnectionStats {
  timeSeries(input: TimeSeriesOptions!, segmentBy: MessageLogSegmentBy): [TimeSeriesBucket!]!
}
```

Reply:
It should be optional from within the TimeSeriesOptions input type as is, with only `bucketDuration` being required

Reply:
The issue isn't that it's optional, but that `TimeSeriesOptions` is generic and meant to be used for more than message logs going forward (otherwise it would be `MessageLogTimeSeriesOptions`), so tying it to `MessageLogSegmentBy` doesn't make sense here.

Reply:
Ah, didn't realize that this is trying to set a precedent- having the option there was confusing to me due to everywhere else mainly using an input object **or** having the input args be comma separated, but I can make this change

Reply:
Pushed these changes, though I don't necessarily like the feel of it.